Control/Monad/Logic/Class.hs

```diff
@@ -165,26 +181,109 @@ class (Monad m, Alternative m) => MonadLogic m where
     (>>-)      :: m a -> (a -> m b) -> m b
     infixl 1 >>-
 
-    -- | Logical __conditional.__ The equivalent of
-    --   <http://lpn.swi-prolog.org/lpnpage.php?pagetype=html&pageid=lpn-htmlse44 Prolog's soft-cut>.
-    --   If its first argument succeeds at all, then the results will be fed into
-    --   the success branch. Otherwise, the failure branch is taken.
-    --   satisfies the following laws:
-    --
-    --   > ifte (pure a) th el       == th a
-    --   > ifte empty th el          == el
-    --   > ifte (pure a <|> m) th el == th a <|> (m >>= th)
-    ifte       :: m a -> (a -> m b) -> m b -> m b
-
     -- | __Pruning.__ Selects one result out of many. Useful for when multiple
     --   results of a computation will be equivalent, or should be treated as
     --   such.
+    --
+    --   As an example, here's a way to determine if a number has an
+    --   factors (whole number divisors):
+    --
+    --   > choose = foldr ((<|>) . pure) empty
+    --   >
+    --   > factors n = do a <- choose [2..n-1]
+    --   >                b <- choose [2..n-1]
+    --   >                guard (a * b == n)
+    --   >                return (a,b)
+    --   >
+    --   > nonPrime_ v = do _ <- factors v
+    --   >                  return "Not Prime"
```

Comment:
My knee-jerk reaction is to rewrite is as `nonPrime_ v = factors v *> pure "Composite"`, but I'm unsure if it makes things easier to comprehend.

Reply:
I agree that your suggestion is the more concise representation and what I would include in project code.   However, here I think that for the purposes of clarity and understandability, especially given some of the comments on the associated issue, that this form is probably more easily understood.